plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/perspective/widgets/machine/appliance/server/ServerProperties.java

```diff
@@ -0,0 +1,46 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A., Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Mario Loriedo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.extension.machine.client.perspective.widgets.machine.appliance.server;
+
+import com.google.inject.Inject;
+import com.google.inject.assistedinject.Assisted;
+
+import org.eclipse.che.api.machine.shared.dto.ServerPropertiesDto;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * ServeProperties class stores non mandatory properties of the current server.
+ *
+ * @author Mario Loriedo
+ */
+public class ServerProperties implements org.eclipse.che.api.core.model.machine.ServerProperties {
+
+    private final ServerPropertiesDto descriptor;
+
+    @Inject
+    public ServerProperties(@Assisted ServerPropertiesDto descriptor) {
+        this.descriptor = descriptor;
+    }
+
+    @Override
+    public String getPath() {
+        return descriptor.getPath();
+    }
+
+    @Override
+    public String getInternalAddress() { return descriptor.getInternalAddress(); }
+
+    @Override
+    public String getInternalUrl() {
+        return descriptor.getInternalUrl();
+    }
```

Comment:
Class https://github.com/eclipse/che/blob/master/plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/perspective/widgets/machine/appliance/server/Server.java had no equals method. Should I add equals/hashcode/toString to existing classes too?